rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleConnection.java

```diff
@@ -417,4 +423,23 @@ public String toString() {
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     Observable<Integer> requestMtu(int mtu);
+
+    /**
+     * <b>This method requires deep knowledge of RxAndroidBLE internals. Use it only as a last resort if you know
+     * what your are doing.</b>
+     * <p>
+     * Queue an operation for future execution. The method accepts a builder parameter. The builder function
+     * will be invoked with three arguments and should return the a non-null instance of {@link RxBleRadioOperation}.
+     * The three arguments the builder receives are:
+     * <ol>
+     * <li>{@link Scheduler} The RxBleRadio scheduler used to execute operation ({@link AndroidSchedulers#mainThread()}</li>
+     * <li>{@link BluetoothGatt} The Android API GATT instance</li>
+     * <li>{@link RxBleGattCallback} The internal Rx ready bluetooth gatt callback to be notified of GATT operations</li>
+     * </ol>
+     *
+     * @param builder The builder used to create the {@link RxBleRadioOperation} instance
+     * @param <T>     The type returned by the {@link RxBleRadioOperation} instance
+     * @return Observable emitting the value after execution or an error in case of failure.
+     */
+    <T> Observable<T> queue(Func3<Scheduler, BluetoothGatt, RxBleGattCallback, RxBleRadioOperation<T>> builder);
```

Comment:
I updated the PR to show the `Observable` idea and I added an example of a custom operation in the sample project.

Now that I think about it, not blocking the main thread might be a problem for people implementing custom operation. For example, let's assume my flashing firmware use case.

If I do `flashFirmareObservable().subcribeOn(scheduler);`, this will block the main thread for the whole flashing progress. To avoid this, I would need to at various interval to let a chance for the main thread to run something like:

```
  // The firmwareChunkWritters would be Observable<Observable<T>>
  return firmwareChunkWritters().concatMap(writter -> writter.subscribeOn(scheduler));
```

We cannot do anything I guess except warning user in the `RxBleRadioOperationCustom` documentation and in the sample.

Anyway, tell me what you think about the `Observable` approach. If you see a problem or you don't like it, will revert back to `observerProxy/releasedAction` way.